### PR TITLE
added SASL EXTERNAL client cert example to both example_test.go files

### PR DESCRIPTION
**Pull request by 62726164**
Hopefully, this pull request will succeed. I've added the SASL EXTERNAL client certificate example to both example_test.go files now. Thanks for closing #274 and #276. This PR is in reference to my initial comment in #273.

### CODE REVIEW

File: examples_test.go [hunk 1]
```diff
@@ -340,3 +342,53 @@ func ExampleControlPaging_manualPaging() {
...
+	// Setup TLS with ldap client cert
+	tlsConfig := &tls.Config{
+		Certificates:       []tls.Certificate{cert},
+		RootCAs:            caCertPool,
+		InsecureSkipVerify: true,
+	}
+	tlsConfig.BuildNameToCertificate()
+
+	// connect to ldap server
+	l, err := ldap.DialURL("ldap://ldap.example.com:389")
```
Comment (johnweldon) [inline examples_test.go:375]: Just need to remove `ldap.` so it reads: `l, err := DialURL(...`

File: v3/examples_test.go [hunk 2]
```diff
@@ -340,3 +342,53 @@ func ExampleControlPaging_manualPaging() {
...
+	// Setup TLS with ldap client cert
+	tlsConfig := &tls.Config{
+		Certificates:       []tls.Certificate{cert},
+		RootCAs:            caCertPool,
+		InsecureSkipVerify: true,
+	}
+	tlsConfig.BuildNameToCertificate()
+
+	// connect to ldap server
+	l, err := ldap.DialURL("ldap://ldap.example.com:389")
```
Comment (johnweldon) [inline v3/examples_test.go:375]: ditto here

**Comment by stefanmcshane** [comment]
Hi @62726164,
It looks like this is failing due to breaking a `go vet` naming convention.

[How to use Example in go](https://blog.golang.org/examples) states that you must use Example with an existing function. As you are doing an External Bind, try renaming your functions to `ExampleConn_ExternalBind` then run `go vet` and `go vet v3/*`

If these succeed, make sure to run the makefile using `make`, and this should pass then!

Let us know if you need any more help!

**Comment by johnweldon** [comment]
the `search.go` and `v3/search.go` error on line 400 is my fault.  You can update your PR by making line 400 use `%s` instead of `%w`  for backward compatibility.

**Comment by 62726164** [comment]
OK, thanks. go vet and make work now. I think when the verb is fixed in the original repo that all checks will pass.

**Comment by johnweldon** [comment]
You should also update the `v3/search.go` to fix the %w verb error.

Also, from my reading of the other error, and digesting the deprecation message on that function, you should just remove line 372 of `examples_test.go` and `v3/examples_test.go`.

**Comment by johnweldon** [comment]
Let me try to fix the `%w` error on master

**Comment by 62726164** [comment]
Thanks for the help @johnweldon and @stefanmcshane. I hope the example helps other users of the library quickly get up and running with TLS client certs.

**Comment by johnweldon** [comment]
Thanks for your patience and contributions @62726164

**Pull request merged**